src/plans/bundle_plan.jl

```diff
@@ -0,0 +1,126 @@
+#
+# Common files for bunlde-based solvers
+#
+
+function bundle_method_subsolver end
+@doc raw"""
+    bundle_method_subsolver(M, bms<:Union{ConvexBundleMethodState, ProximalBundleMethodState})
```

Comment:
Could we change the second argument to be a little less dependent on the actual state?

I imagine, this could also be something one could implemented given a bundle of points and vectors. So that it would be maybe something like `bundle_method_subsolver(M, bundle)`

Sure for the two different bundle methods we would no longer dispatch on the state but implement two functions with different names.